### ISSUE TITLE
Running a flow which imports Polars in a docker-compose environment results in a sigkill signal

### ISSUE THREAD

**Issue by Jurtracey** [PrefectHQ/prefect#7622]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Submitting a flow run in a docker-compose environment which imports [polars](https://www.pola.rs/) (python) results in a sigkill signal for the flow. 

### Reproduction

```python3
--- Docker compose setup ---
version: "3.9"

networks:
  default:
    name: prefect-network

volumes:
  prefect:
  db:
  minio:

services:
  ### Prefect Database
  database:
    image: postgres:14.1-alpine
    restart: always
    environment:
      - POSTGRES_USER=
      - POSTGRES_PASSWORD=
      - POSTGRES_DB=orion
    expose:
      - 5432
    volumes:
      - db:/var/lib/postgresql/data

  ### MinIO for flow storage
  minio:
    image: minio/minio:latest
    entrypoint: [ "minio", "server", "--address", "0.0.0.0:9000", "--console-address", "0.0.0.0:9001", "/data" ]
    volumes:
      - "minio:/data"
    ports:
      - "9000:9000"
      - "9001:9001"

  ### Prefect Orion API
  orion:
    image: prefecthq/prefect:2.6.8-python3.11
    restart: always
    entrypoint: ["prefect", "orion", "start"]
    volumes:
      - prefect:/root/.prefect
    environment:
      PREFECT_ORION_API_HOST: 0.0.0.0
      PREFECT_ORION_DATABASE_CONNECTION_URL: postgresql+asyncpg://postgres:postgres@database:5432/orion
    ports:
      - "4200:4200"
    depends_on:
      - database

  ## Prefect Agent
  agent:
    image: prefecthq/prefect:2.6.8-python3.11
    restart: always
    command: sh -c "pip install --upgrade pip && pip install polars s3fs && prefect agent start -q main"
    environment:
      PREFECT_API_URL: http://orion:4200/api
      PREFECT_LOGGING_LEVEL: DEBUG

--- Prefect flow ---

from prefect import flow
import polars as pl


@flow(name="run_test_polars", retries=0)
def run_test_polars():
    pass

--- Flow deploy ---

deployment = await Deployment.build_from_flow(
            flow=run_test_polars,
            name="run_test_polars",
            version=1,
            work_queue_name="main",
            storage=storage,
            ignore_file=".prefectignore",
        )
```

### Error

```python3
2022-11-22 15:04:44 14:04:44.421 | INFO    | prefect.agent - Submitting flow run '94788372-fc87-4dba-a6fd-6c7fc1dd759d'
2022-11-22 15:04:44 14:04:44.663 | INFO    | prefect.infrastructure.process - Opening process 'slim-wallaby'...
2022-11-22 15:04:44 14:04:44.672 | DEBUG   | prefect.infrastructure.process - Process 'slim-wallaby' running command: /usr/local/bin/python -m prefect.engine in /tmp/tmp1x9ta81rprefect
2022-11-22 15:04:44 14:04:44.735 | INFO    | prefect.agent - Completed submission of flow run '94788372-fc87-4dba-a6fd-6c7fc1dd759d'
2022-11-22 15:04:49 14:04:49.428 | DEBUG   | prefect.agent - Checking for flow runs...
2022-11-22 15:04:52 14:04:52.615 | DEBUG   | prefect.client - Connecting to API at http://orion:4200/api/
2022-11-22 15:04:53 14:04:53.844 | DEBUG   | Flow run 'slim-wallaby' - Loading flow for deployment 'run_test_polars'...
2022-11-22 15:04:53 <jemalloc>: MADV_DONTNEED does not work (memset will be used instead)
2022-11-22 15:04:53 <jemalloc>: (This is the expected behaviour if you are running under QEMU)
2022-11-22 15:05:08 14:05:08.374 | ERROR   | prefect.infrastructure.process - Process 'slim-wallaby' exited with status code: -9; This indicates that the process exited due to a SIGKILL signal. Typically, this is caused by high memory usage causing the operating system to terminate the process.
```

### Versions

```Text
prefect agent container:
14:06:15.570 | DEBUG   | prefect.client - Connecting to API at http://orion:4200/api/
Version:             2.6.8
API version:         0.8.3
Python version:      3.11.0
Git commit:          68044e28
Built:               Thu, Nov 17, 2022 3:19 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         hosted

prefect orion container:
Version:             2.6.8
API version:         0.8.3
Python version:      3.11.0
Git commit:          68044e28
Built:               Thu, Nov 17, 2022 3:19 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         ephemeral
Server:
  Database:          postgresql
```

**Comment by madkinsz** [comment]
Interesting! We'll need to determine the root cause of the SIGKILL before we can do anything here.

**Comment by Jurtracey** [comment]
Other flows in this setup which do not require polars can be run, scheduled normally. Do you have any suggestions for me to look further into this issue? Is there any information that I can add to this post which helps finding the cause of this issue?

**Comment by Jurtracey** [comment]
Upon further researching this issue and checking the docker images this seems to be related to using AMD64 images on MacOS. Docker discourages this which I did not know. Running the prefect agent on a non-AMD64 python image with prefect and polars installed manually exits without issues. I will close this issue as resolved.

I guess this slightly ties in with [#5388](https://github.com/PrefectHQ/prefect/issues/5388) which is looking into support for ARM based prefect images which would be nice to have.

Thanks for the response.

**Comment by madkinsz** [comment]
Thanks for looking into it! Glad you got past it.
